svgdigitizer/electrochemistry/cv.py

```diff
@@ -18,111 +18,166 @@
+                'y': {'dimension': 'j' if 'm2' in str(self.get_axis_unit('y')) else 'I',
+                      'unit': 'A / m2' if 'm2' in str(self.get_axis_unit('y')) else 'A'}}
 
-        # TODO: should be j if both current and normalized to are given in the yaml file
-        self.ylabel = 'I'
+    def get_axis_unit(self, axis):
+        r'''
+        Replaces the units derived from the svg file into astropy units.
+        '''
+        unit = self.svgplot.units[axis]
 
-        # TODO: All the rest in the init is presumably not necessary
-        self.description = self.metadata['figure description']
+        return self.get_correct_unit(unit)
 
-        self.xunit = self.description['potential scale']['unit']
+    @classmethod
+    def get_correct_unit(cls, unit):
+        for correct_unit, typos in unit_typos.items():
+            if unit in typos:
+                return u.Unit(correct_unit)
 
-        self.yunit = self.description['current']['unit']
+        raise ValueError(f'Unknown Unit {unit} on Axis {axis}')
 
-        self.get_rate()
+    @property
+    @cache
+    def rate(self):
+        r'''
+        Return a rate based on a label in the SVG file.
+        '''
+        rates = self.svgplot.svg.get_texts('(?:scan rate|rate): (?P<value>-?[0-9.]+) *(?P<unit>.*)')
+        # To Do
+        # assert that only one label contains the scan rate
+        # asstert that a rate is available at all
 
-        self.modify_df()
+        # Convert to astropy unit
+        rates[0].unit = CV.get_correct_unit(rates[0].unit)
+        # Convert to SI astropy unit: V / s
+        rate = float(rates[0].value) * rates[0].unit
 
-    def get_rate(self):  # TODO: probably not required
-        r'''
-        Return rate based on the x coordinate units.
+        return rate
 
-        At the moment we simply use the value.
-        '''
-        self.rate = self.description['scan rate']['value']
-        return self.rate
+    @property
+    @cache
+    def cv_df(self):
+        # Create potential column.
+        df = self.create_df_U_axis(self.svgplot.df[['x']])
+
+        # Create current or current density column.
+        df = pd.concat([df, self.create_df_I_axis(self.svgplot.df[['y']])], axis=1)
 
-    def modify_df(self):
-        # Create potential columns
-        self.df = self.create_df_U_axis(self.svgplot.dfs[0])
-        # Create current columns
-        self.df = pd.concat(
-            [self.df, self.create_df_I_axis(self.svgplot.dfs[0])],
-            axis=1)
+        # Create time axis.
+        df['t'] = self.create_df_time_axis(df)
+        # Rearrange columns.
+        df = df[['t', 'U', self.axis_properties['y']['dimension']]]
 
-        # create time axis
-        self.df['t'] = self.create_df_time_axis(self.svgplot.dfs[0])
+        return df
 
     def create_df_U_axis(self, df):
         r'''
         Create voltage axis in the dataframe based on the units given in the
         figure description.
```

Comment:
In all of these docstrings, that this comes from the figure description is not really true anymore. It's maybe better to not be explicit about this and refer the reader to the correct place, i.e., just say: ``Add voltage column to the dataframe `df`, based on the :meth:`get_axis_unit`.``